src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculator.php

```diff
@@ -50,29 +43,30 @@ public function calculateOriginal(ProductVariantInterface $productVariant, array
         $channelPricing = $productVariant->getChannelPricingForChannel($context['channel']);
 
         if (null === $channelPricing) {
-            throw new MissingChannelConfigurationException(sprintf(
-                'Channel %s has no price defined for product variant %s',
-                $context['channel']->getName(),
-                $productVariant->getName()
-            ));
+            throw new MissingChannelConfigurationException($this->getMissingChannelProductVariantPriceMessage($context['channel'], $productVariant));
         }
 
         if (null === $channelPricing->getOriginalPrice()) {
-            if ($channelPricing->getPrice() === null) {
-                $message = sprintf('Channel %s has no price defined for product variant', $context['channel']->getName());
-
-                if ($productVariant->getName() !== null) {
-                    $message .= sprintf(' %s (%s)', $productVariant->getName(), $productVariant->getCode());
-                } else {
-                    $message .= sprintf(' with code %s', $productVariant->getCode());
-                }
-
-                throw new MissingChannelConfigurationException($message);
+            if ($channelPricing->getPrice() !== null) {
+                return $channelPricing->getPrice();
             }
 
-            return $channelPricing->getPrice();
+            throw new MissingChannelConfigurationException($this->getMissingChannelProductVariantPriceMessage($context['channel'], $productVariant));
         }
 
         return $channelPricing->getOriginalPrice();
     }
+
+    private function getMissingChannelProductVariantPriceMessage(ChannelInterface $channel, ProductVariantInterface $productVariant): string
+    {
+        $message = sprintf('Channel %s has no price defined for product variant', $channel->getName());
+
+        if ($productVariant->getName() !== null) {
```

Comment:
You can remove the `else` here and return early, like:
```suggestion
        if ($productVariant->getName() !== null) {
            return $message . sprintf(' %s (%s)', $productVariant->getName(), $productVariant->getCode());
        }

        return $message . sprintf(' with code %s', $productVariant->getCode());
```